lib/resources/os.rb

```diff
@@ -27,15 +27,33 @@ class OSResource < Inspec.resource(1)
       end
     end
 
+    # This is a string override for os.name.
+    # TODO: removed in inspec 2.0
+    class NameCleaned < String
+      def ==(other)
+        if other =~ /[A-Z ]/
+          cleaned = other.downcase.tr(' ', '_')
+          Inspec::Log.warn "[DEPRECATED] Platform names will become lowercase in InSpec 2.0. Please match on '#{cleaned}' instead of '#{other}'"
+        end
+        super(other)
+      end
+    end
+
     def [](name)
       # convert string to symbol
       name = name.to_sym if name.is_a? String
+      return self.name if name == :name
```

Comment:
For clarity purposes, we should probably change the method parameter from `name` to `key` or something like that so we're not overloading the use of `name` here.